src/inmanta/agent/agent.py

```diff
@@ -1103,6 +1109,8 @@ def __init__(
         self.hostname = hostname
         self.poolsize = poolsize
         self.ratelimiter = asyncio.Semaphore(poolsize)
+        # Number of in flight requests for resolving CAD's
+        self.cad_ratelimiter = asyncio.Semaphore(3)
```

Comment:
total connection pool is 10 
I could go higher here

Reply:
I am not sure what is best here. But 3 sounds reasonable to me.

Reply:
I really don't have a view on what's sensible here.